specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/searchservice.json

```diff
@@ -7550,6 +7808,34 @@
       ],
       "description": "Represents a search index definition, which describes the fields and search behavior of an index."
     },
+    "SearchAlias": {
+      "properties": {
+        "name": {
+          "externalDocs": {
+            "url": "https://docs.microsoft.com/rest/api/searchservice/Naming-rules"
+          },
+          "type": "string",
+          "description": "The name of the alias."
+        },
+        "indexes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The name of the index this alias maps to. Only one index name may be specified."
```

Comment:
Is the comment outdated? Why does it take in an array?

Reply:
This is a common pattern for services who want to future-proof their APIs. An array that only accepts a single value today may accept multiple values tomorrow and that would not be a breaking change, but changing from a single type to an array would be.

Reply:
Travis is spot on. Eventually we may want to support customers being able to specify multiple indexes for a single alias.

Reply:
Another use case - multiple aliases for a single index. Is that a less common scenario?

Reply:
Thanks for bringing those up. My take on the your questions:
> Just wondering if the comment should be clearer to say something like "Only one index name may be specified currently".

The *currently* would imply we have plans to add it in the future. Since we aren't sure of that, I think it's best to omit it.

> Also, I hope we update the comment when in a later update the service behavior changes.

Yes, we will update it in the future if/when the behavior changes.

> Another use case - multiple aliases for a single index. Is that a less common scenario?

This is allowed. The customer would just need to create multiple aliases, each mapping to the same index.